src/server/scheduler.rs

```diff
@@ -0,0 +1,406 @@
+                    }),
+                Resource::Light(_light) => {
+                    vec![WakeupRequest::Light(resource_link)]
+                }
+                Resource::BridgeHome(_bridge_home) => {
+                    let all_rooms = lock.get_resources_by_type(RType::Room);
+                    all_rooms
+                        .into_iter()
+                        .filter_map(|room_resource| match room_resource.obj {
+                            Resource::Room(room) => {
+                                let grouped_light = room.grouped_light_service()?;
+                                Some(WakeupRequest::Group(*grouped_light))
+                            }
+                            _ => None,
+                        })
+                        .collect()
+                }
+                _ => Vec::new(),
+            })
+            .collect::<Vec<_>>()
+    };
+
+    for request in &requests {
+        if let Err(err) = request.on(res.clone(), config.clone()).await {
+            log::warn!("Failed to turn on wake up light: {}", err);
+        }
+    }
+
+    // wait until fade in has completed
+    // otherwise the behavior instance can be disabled before it has actually finished
+    sleep(config.fade_in_duration.to_std()).await;
+
+    if let Some(duration) = config.turn_lights_off_after {
+        sleep(duration.to_std()).await;
+
+        for request in &requests {
+            if let Err(err) = request.off(res.clone()).await {
+                log::warn!("Failed to turn off wake up light: {}", err);
+            }
+        }
+    }
+}
+
+enum WakeupRequest {
+    Light(ResourceLink),
+    Group(ResourceLink),
+}
+
+impl WakeupRequest {
+    async fn on(&self, res: Arc<Mutex<Resources>>, config: WakeupConfiguration) -> ApiResult<()> {
+        let light_supports_effects = match self {
+            Self::Light(resource_link) => res
+                .lock()
+                .await
+                .get::<Light>(resource_link)?
+                .effects
+                .is_some(),
+            Self::Group(_) => false, // todo: implement when grouped light support effects
+        };
+        let use_sunrise_effect =
+            light_supports_effects && config.style == Some("sunrise".to_string());
+
+        if use_sunrise_effect {
+            self.sunrise_on(&res, &config).await?;
+        } else {
+            self.transition_to_bright_on(res, &config).await?;
+        }
+        Ok(())
+    }
+
+    async fn transition_to_bright_on(
+        &self,
+        res: Arc<Mutex<Resources>>,
+        config: &WakeupConfiguration,
+    ) -> Result<(), crate::error::ApiError> {
+        // As reported by the Hue bridge
+        const WAKEUP_FADE_MIREK: u16 = 447;
+
+        // Reset brightness and set color temperature
+        let reset_backend_request = match self {
+            Self::Light(resource_link) => {
+                let payload = LightUpdate::default()
+                    .with_on(Some(On::new(true)))
```

Comment:
If you like you could put this under `impl On { ...` in `crates/hue/api/light.rs`:

```rust
impl From<bool> for On {
    fn from(value: bool) -> Self {
        Self::new(value)
    }
}
```

Then we could write:

```rust
   .with_on(Some(true))
```